spec/gamelogic_spec.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+require './lib/gamelogic.rb'
+
+RSpec.describe '#initialize' do
+  before(:each) do
+    @game = TicTacToe.new
+  end
+
+  it 'should initialize the board with array of spaces and set variables' do
+    expect(@game.board[0...9]).to eq([' ', ' ', ' ', ' ', ' ', ' ', ' ', ' ', ' '])
+  end
+
+  it "shouldn't initialize the board with an empty array" do
+    expect(@game.board[0...8]).not_to eq(['', '', '', '', '', '', '', '', ''])
+  end
+
+  it "should set the 'player' variable to 'x'" do
+    expect(@game.player).to eq('x')
+  end
+
+  it "shouldn't set the 'player' variable to 'o'" do
+    expect(@game.player).not_to eq('o')
+  end
+end
+
+RSpec.describe '#check_win_aux' do
+  before(:each) do
+    @game = TicTacToe.new
+  end
+
+  it 'should not declare a winner with an empty board' do
+    @game.board[0...2] = [' ', ' ', ' ']
+    expect(@game.check_win_aux([0, 1, 2])).to eq(nil)
+  end
+
+  it 'should declare a winner if a row of 3 is passed' do
+    @game.board[0...2] = %w[x x x]
+    expect(@game.check_win_aux([0, 1, 2])).to eq('x')
+  end
+
+  it 'should not declare winner without a row of 3' do
+    @game.board[0...2] = %w[x o x]
+    expect(@game.check_win_aux([0, 1, 2])).to eq(nil)
+  end
+end
```

Comment:
I am curious why you are using `0...2` here. Do you remember this translates to `0, 1` (ie: exclusive range)? Your array size increases by 1 after that line. This means you designed your board to overflow and there is no check against it. :thinking: **That's very bad!**

Also, you should not update your board that way. There should be only one to update the board (via #make_move). That's why you made the board read-only in the first place. Also, it would be very testable and bug-free. Please fix this.